### PR TITLE
Extract operation name and type from execution context

### PR DESCRIPTION
**Pull request by jkimbo**
## Description

Add properties to ExecutionContext to extract the operation name and type from the graphql document.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Jonathan Kim for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: strawberry/types/execution.py [hunk 1]
```diff
@@ -1,37 +1,97 @@
...
     context: Any = None
     variables: Optional[Dict[str, Any]] = None
-    operation_name: Optional[str] = None
     root_value: Optional[Any] = None
     validation_rules: Tuple[Type[ASTValidationRule], ...] = dataclasses.field(
         default_factory=lambda: tuple(specified_rules)
     )
 
+    # The operation name that is provided by the request
+    _provided_operation_name: Optional[str] = None
```
[the remark this answers is not in the document]
Comment (BryceBeagle) [reply strawberry/types/execution.py:34]: Maybe we can use [`InitVar`](https://docs.python.org/3/library/dataclasses.html#init-only-variables)

File: strawberry/types/execution.py [hunk 2]
```diff
@@ -1,37 +1,97 @@
...
+            raise RuntimeError("No GraphQL document available")
+
+        definition = next(
+            (
+                node
+                for node in graphql_document.definitions
+                if isinstance(node, OperationDefinitionNode)
+            ),
+            None,
+        )
```
Comment (BryceBeagle) [inline strawberry/types/execution.py:92]: Above, in the `operation_type` properly, you used a for loop with a `break` for a very similar operation. Suggest using the same technique for both

**Review by BryceBeagle — CHANGES_REQUESTED** [review]

File: strawberry/types/execution.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jkimbo) [reply strawberry/types/execution.py:34]: Yep agreed.

**Comment by jkimbo** [comment]
@BryceBeagle @patrick91 this is ready for review again.

**Review by patrick91 — APPROVED** [review]

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Release type: patch
```
Comment (patrick91) [inline RELEASE.md:1]:
```suggestion
Release type: minor
```

**Pull request merged**